project.py

```diff
@@ -114,6 +134,79 @@ def create_trackbars(self):
         cv.createTrackbar('lo', 'result', self.lo, 255, self.trackbar_callback)
         cv.createTrackbar('hi', 'result', self.hi, 255, self.trackbar_callback)
 
+    def check_change(self, image, mask):
+        imageOld = cv.imread("slide"+repr(self.numberOfSlides)+".png",
+                                cv.CV_LOAD_IMAGE_COLOR)
+        maskOld = cv.imread("mask.png", cv.CV_LOAD_IMAGE_GRAYSCALE)
+        if(maskOld is None):
+            return True
+
+        if(mask.shape != maskOld.shape):
+            return False
+
+        applyMask = cv.bitwise_and(maskOld, mask)
+        imageOld = cv.bitwise_and(imageOld, imageOld, mask=applyMask)
+        checkImage = cv.bitwise_and(image, image, mask=applyMask)
+
+        orb = cv.ORB()
+        kp1, ds1 = orb.detectAndCompute(imageOld, None)
+        kp2, ds2 = orb.detectAndCompute(checkImage, None)
+        bf = cv.BFMatcher(cv.NORM_HAMMING, crossCheck=True)
+        matches = bf.match(ds1, ds2)
+
+        val = float(len(matches))/len(kp1)
+        if(val > self.similarity):
+            print "Is the same"
+
+        return False
+
+    def get_sorted_rectangle(self, cnt, image):
+        pts = cnt.reshape(4, 2)
+        rect = np.zeros((4, 2), dtype="float32")
+
+        s = pts.sum(axis=1)
+        rect[0] = pts[np.argmin(s)]
+        rect[2] = pts[np.argmax(s)]
+
+        diff = np.diff(pts, axis=1)
+        rect[1] = pts[np.argmin(diff)]
+        rect[3] = pts[np.argmax(diff)]
+
+        return rect
+
+    def get_croped_image(self, rect, image, mask):
```

Comment:
@mrshu Yes I did. It's because openCV doesn't have function that will crop image based on conture and I'm also doing in that function perspective wraping